src/include/binder/expression/parameter_expression.h

```diff
@@ -11,7 +11,8 @@ class ParameterExpression : public Expression {
     explicit ParameterExpression(
         const std::string& parameterName, std::shared_ptr<common::Value> value)
         : Expression{common::ExpressionType::PARAMETER,
-              common::LogicalType(common::LogicalTypeID::ANY), createUniqueName(parameterName)},
+              common::LogicalType(value->getDataType()->getLogicalTypeID()),
```

Comment:
this is not safe because logical type can be nested. Just do `common::LogicalType(value->getDataType())`